Realm/RLMObject.mm

```diff
@@ -219,3 +219,31 @@ - (void)setValue:(id)value forUndefinedKey:(NSString *)key {
 }
 
 @end
+
+@implementation RLMWeakObjectHandle {
+    realm::Row _row;
+    RLMRealm *_realm;
+    RLMObjectSchema *_objectSchema;
+    Class _objectClass;
+}
+
+- (instancetype)initWithObject:(RLMObjectBase *)object {
+    if (!(self = [super init])) {
+        return nil;
+    }
+
+    _row = object->_row;
+    _realm = object->_realm;
+    _objectSchema = object->_objectSchema;
+    _objectClass = object.class;
```

Comment:
For `LinkingObjects` purposes you're right that `RLMObjectBase` is sufficient. I went with returning the object's class as it seemed more correct, and avoids any potential problems if in the future the objects returned by this class make their way back to user code. It doesn't seem like it's expensive enough relative to the other work that happens initializing Swift generics to worry about for performance reasons.